File: AmanRajSingh/BinarySearch/SumofSquareNumbers.java
class Solution {
    public boolean judgeSquareSum(int c) {
        long start = 0;
        long end = (long )Math.sqrt(c);
        long mid = 0;

        while(start <= end){
            mid = start * start + end * end;
            if(mid == c){
                return true;
            }else if(mid < c) {
               start++;
            }else{
                end--;
            }
        }
        return false;
    }
}

File: AdityaKumar1010707/BinarySearch/sumOfSquareNumbers.cpp
class Solution {
    public:
        bool judgeSquareSum(int c) {
    
    
            int left=0;
            int right=sqrt(c);
            while(left<=right){
    
                long long leftSquareVal = (long long)left*left;
                long long rightSquareVal = (long long)right*right;
    
    
                long long val=leftSquareVal + rightSquareVal;
    
                if(val==c)return true;
                else if(val>c)right--;
                else  left++;
    
            }
    
            return false;
            
        }
    };